builtin/logical/nomad/path_creds_create.go

```diff
@@ -0,0 +1,79 @@
+package nomad
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func pathCredsCreate(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "creds/" + framework.GenericNameRegex("name"),
+		Fields: map[string]*framework.FieldSchema{
+			"name": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Name of the role",
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation: b.pathTokenRead,
+		},
+	}
+}
+
+func (b *backend) pathTokenRead(
+	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	name := d.Get("name").(string)
+
+	role, err := b.Role(req.Storage, name)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving role: %s", err)
+	}
+	if role == nil {
+		return logical.ErrorResponse(fmt.Sprintf("Role '%s' not found", name)), nil
+	}
+
+	// Determine if we have a lease configuration
+	leaseConfig, err := b.LeaseConfig(req.Storage)
+	if err != nil {
+		return nil, err
+	}
+	if leaseConfig == nil {
+		leaseConfig = &configLease{}
+	}
+
+	// Get the nomad client
+	c, err := b.client(req.Storage)
+	if err != nil {
+		return nil, err
+	}
+
+	// Generate a name for the token
+	tokenName := fmt.Sprintf("Vault %s %s %d", name, req.DisplayName, time.Now().UnixNano())
+
+	// Create it
+	token, _, err := c.ACLTokens().Create(&api.ACLToken{
+		Name:     tokenName,
+		Type:     role.TokenType,
+		Policies: role.Policy,
+		Global:   role.Global,
+	}, nil)
+	if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
```

Comment:
`logical.ErrorResponse` is used for user errors. Can we make it `return nil, err`?

Reply:
Yes, fixed in a5f01d49e.